test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetPathOption.java

```diff
@@ -0,0 +1,225 @@
+                               """);
+        String snippetPathValue = Stream.of("directoryA", "directoryB")
+                .map(src::resolve)
+                .map(Path::toAbsolutePath)
+                .map(Path::toString)
+                .collect(Collectors.joining(File.pathSeparator));
+        javadoc("-d", base.resolve("out").toString(),
+                "--snippet-path", snippetPathValue,
+                "-sourcepath", src.toString(),
+                "pkg");
+        checkExit(Exit.OK);
+        checkOutput("pkg/X.html", true, "Hello, snippet-files!");
+    }
+
+    @Test
+    public void test3(Path base) throws Exception {
+        Path src = Files.createDirectories(base.resolve("src"));
+        tb.createDirectories(src.resolve("directoryA"), src.resolve("directoryB"));
+        tb.writeFile(src.resolve("directoryA/mysnippet.txt"), "Hello, directoryA!");
+        tb.writeFile(src.resolve("directoryB/mysnippet.txt"), "Hello, directoryB!");
+        tb.writeJavaFiles(src, """
+                               package pkg;
+
+                               /** {@snippet file="mysnippet.txt"} */
+                               public class X { }
+                               """);
+        String snippetPathValue = Stream.of("directoryA", "directoryB")
+                .map(src::resolve)
+                .map(Path::toAbsolutePath)
+                .map(Path::toString)
+                .collect(Collectors.joining(File.pathSeparator));
+        javadoc("-d", base.resolve("out").toString(),
+                "--snippet-path", snippetPathValue,
+                "-sourcepath", src.toString(),
+                "pkg");
+        checkExit(Exit.OK);
+        checkOutput("pkg/X.html", true, "Hello, directoryA!");
+        checkOutput("pkg/X.html", false, "Hello, directoryB!");
+    }
+
+    @Test
+    public void test4(Path base) throws Exception {
+        Path src = Files.createDirectories(base.resolve("src"));
+        tb.writeJavaFiles(src, """
+                               package pkg;
+
+                               /** {@snippet file="mysnippet.txt"} */
+                               public class X { }
+                               """);
+        String snippetPathValue = Stream.of("directoryA", "directoryB")
+                .map(src::resolve)
+                .map(Path::toAbsolutePath)
+                .map(Path::toString)
+                .collect(Collectors.joining(File.pathSeparator));
+        javadoc("-d", base.resolve("out").toString(),
+                "--snippet-path", snippetPathValue,
+                "-sourcepath", src.toString(),
+                "pkg");
+        checkExit(Exit.ERROR);
+    }
+
+    /*
+     * Tests that the elements of the snippet path are iteratively searched
+     * until the file is found. In particular, tests that if the file is not
+     * immediately found, the search is not abandoned.
+     */
+    @Test
+    public void testSearchPath(Path base) throws Exception {
```

Comment:
Not wrong, but arguably somewhat superfluous. This is close to being a file manager test, not a snippet test.

Reply:
I can add a comment that this test is a part of an end-to-end demonstration of how --snippet-path should work. This is a black-box test that assumes nothing about the implementation of the feature.

Reply:
OK, but I still think the test is somewhat unnecessary. But now you have it, don't delete it.